api/include/opentelemetry/common/spin_lock_mutex.h

```diff
@@ -23,6 +37,15 @@ class SpinLockMutex
   SpinLockMutex &operator=(const SpinLockMutex &) = delete;
   SpinLockMutex &operator=(const SpinLockMutex &) volatile = delete;
 
+  /**
+   * Attempts to lock the mutex.  Return immediately with `true` (success) or `false` (failure).
+   */
+  bool try_lock() noexcept
+  {
+    return !flag_.load(std::memory_order_relaxed) &&
```

Comment:
Is this `.load` needed? Seems just on call to `.exchange` is sufficient for `try_lock`?

Reply:
It's to help with cache miss issues.

See this blog which changed my mind on it: 
https://rigtorp.se/spinlock/